src/miner.h

```diff
@@ -18,18 +18,22 @@ class CWallet;
 
 struct CBlockTemplate;
 
-/** Run the miner threads */
-void GenerateBitcoins(bool fGenerate, CWallet* pwallet, int nThreads);
 /** Generate a new block, without valid proof-of-work */
 CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn, CWallet* pwallet, bool fProofOfStake);
-CBlockTemplate* CreateNewBlockWithKey(CReserveKey& reservekey, CWallet* pwallet, bool fProofOfStake);
 /** Modify the extranonce in a block */
 void IncrementExtraNonce(CBlock* pblock, CBlockIndex* pindexPrev, unsigned int& nExtraNonce);
 /** Check mined block */
 void UpdateTime(CBlockHeader* block, const CBlockIndex* pindexPrev);
 
-void BitcoinMiner(CWallet* pwallet, bool fProofOfStake);
-void ThreadStakeMinter();
+#ifdef ENABLE_WALLET
+    /** Run the miner threads */
+    void GenerateBitcoins(bool fGenerate, CWallet* pwallet, int nThreads);
+    /** Generate a new block, without valid proof-of-work */
+    CBlockTemplate* CreateNewBlockWithKey(CReserveKey& reservekey, CWallet* pwallet, bool fProofOfStake);
+
+    void BitcoinMiner(CWallet* pwallet, bool fProofOfStake);
+    void ThreadStakeMinter();
+#endif // ENABLE_WALLET
```

Comment:
Good call moving these functions under `ENABLE_WALLET` condition. Did you check they aren't called anywhere without the condition in the code?

Reply:
Yes; `BitcoinMiner` is only called from `ThreadBitcoinMiner()` and `ThreadStakeMinter()` both of which require the `ENABLE_WALLET` definition; as does the instantiation of those two threads.

It actually looks like a common issue; which I presume just causes warnings if build without ENABLE_WALLET, but I have been planning on running that test at some point when my build system is actually idle to really see how off it is.  

(Probably should really be a travis regression test; build everything without ENABLE_WALLET and see what explodes)

Reply:
Yeah, I guess we'll have to add such a travis case sooner or later. That's out of scope here.